util/pace/util/initialization/allocator.py

```diff
@@ -1,7 +1,9 @@
-from typing import Callable, Sequence
+from typing import Callable, Optional, Sequence
 
 import numpy as np
 
+from pace.util.halo_data_transformer import QuantityHaloSpec
+
 from .._optional_imports import gt4py
 from ..constants import SPATIAL_DIMS, X_DIMS, Y_DIMS, Z_DIMS
 from ..quantity import Quantity
```

Comment:
Is there a reason the added line is an absolute import?

Reply:
Because it was added automatically by VS Code. I'll clean it up when fixing the failing physics tests, but we generally don't worry about these.